### PR TITLE
maven checkstyle refactoring

### PR DESCRIPTION
**Pull request by alexff91**
There's only stratosphere-java refactored for now.

### CODE REVIEW

File: stratosphere-java/src/main/java/eu/stratosphere/api/java/operators/JoinOperator.java [hunk 1]
```diff
@@ -144,220 +146,222 @@ protected EquiJoin(DataSet<I1> input1, DataSet<I2> input2,
...
+
 	public static final class DefaultJoin<I1, I2> extends EquiJoin<I1, I2, Tuple2<I1, I2>> {
 
-		protected DefaultJoin(DataSet<I1> input1, DataSet<I2> input2, 
+		protected DefaultJoin(DataSet<I1> input1, DataSet<I2> input2,
 				Keys<I1> keys1, Keys<I2> keys2, JoinHint hint)
 		{
 			super(input1, input2, keys1, keys2,
-				(JoinFunction<I1, I2, Tuple2<I1, I2>>) new DefaultJoinFunction<I1, I2>(),
+				new DefaultJoinFunction<I1, I2>(),
```
Comment (rmetzger) [inline stratosphere-java/src/main/java/eu/stratosphere/api/java/operators/JoinOperator.java]: Oh, why did Eclipse remove the cast here?

Comment (alexff91) [inline stratosphere-java/src/main/java/eu/stratosphere/api/java/operators/JoinOperator.java]: Hmm it removes unnecesary casting. Should I return this casting?
28.03.2014 11:26 ÐÏÌØÚÏ×ÁÔÅÌØ "Robert Metzger" notifications@github.com
ÎÁÐÉÓÁÌ:

Comment (rmetzger) [inline stratosphere-java/src/main/java/eu/stratosphere/api/java/operators/JoinOperator.java]: Sorry, I don't know the details about this case. I would suggest leave it this way (so don't change the pull request until somebody has justified arguments)

Comment (alexff91) [inline stratosphere-java/src/main/java/eu/stratosphere/api/java/operators/JoinOperator.java]: Ok, i will leave this and another casting.

**Comment by rmetzger** [comment]
Okay @alexff91, I'm quite happy with your pull request!

There is only one thing that might be a problem: Your change contains a lot of cases where you remove the tabs in empty lines. In general, it is very good to remove them. The problem I'm seeing is that your change will probably break a lot of the other pending pull requests.
Can you investigate if there is a way to avoid these replacements? If yes, avoid them and finish the pull request (by doing the change for all the other modules as well.)

Otherwise (if you do not find a way to avoid it), I would like to have at least one other opinion from another core developer.

**Comment by alexff91** [comment]
I will try to fix this problem with replacements.
29.03.2014 16:15 ÐÏÌØÚÏ×ÁÔÅÌØ "Robert Metzger" notifications@github.com
ÎÁÐÉÓÁÌ:

**Comment by alexff91** [comment]
Hi @rmetzger I've done. I used AnyEdit Plugin for removing spaces if interesting. If everything is okey, what I should do? Keep on refactoring for all other modules?

**Comment by rmetzger** [comment]
Nice. Your changes look very good. I think you can apply them to all modules and I will merge your change.

**Comment by alexff91** [comment]
Okey

On Mon, Mar 31, 2014 at 12:35 PM, Robert Metzger
notifications@github.comwrote:

> Nice. Your changes look very good. I think you can apply them to all
> modules and I will merge your change.
> 
> ## 
> 
> Reply to this email directly or view it on GitHubhttps://github.com/stratosphere/stratosphere/pull/636#issuecomment-39064899
> .

## 

With best wishes and regards,
Fedorov Aleksandr.

**Comment by alexff91** [comment]
Hi @rmetzger ! I've refactored the whole project, tell me if there is any problems.

**Comment by rmetzger** [comment]
manually triggered a build https://travis-ci.org/rmetzger/stratosphere/builds/22109234

**Comment by alexff91** [comment]
Ok, I'll fix it soon.
02.04.2014 18:44  "Robert Metzger" notifications@github.com:

**Comment by rmetzger** [comment]
you have to build using "mvn clean verify -Dhadoop.profile=2" to include the yarn (and hbase) project.

**Comment by rmetzger** [comment]
triggered manual build: https://travis-ci.org/rmetzger/stratosphere/builds/22126486

**Comment by alexff91** [comment]
Oh, there was a problem, when I run "mvn clean verify -Dhadoop.profile=2" I got a error, I pushed to see will be the same error in travis:
The following artifacts could not be resolved: org.apache.hadoop:hadoop-yarn-client:jar:1.2.1, org.apache.hadoop:hadoop-common:jar:1.2.1, org.apache.hadoop:hadoop-hdfs:jar:1.2.1, org.apache.hadoop:hadoop-mapreduce-client-core:jar:1.2.1: Failure to find org.apache.hadoop:hadoop-yarn-client:jar:1.2.1 in http://repo.maven.apache.org/maven2 was cached in the local repository, resolution will not be reattempted until the update interval of central has elapsed or updates are forced

**Comment by rmetzger** [comment]
in which directory did you execute the "clean verify" ?

**Comment by alexff91** [comment]
in "stratosphere" directory with reactor pom

**Comment by alexff91** [comment]
I have a build success locally. Travis is started on https://travis-ci.org/alexff91/stratosphere/builds/22133955.

**Comment by rmetzger** [comment]
Ok. I'll test it tomorrow

Sent from my iPhone

On 02.04.2014, at 22:31, Aleksandr notifications@github.com wrote:

I have a build success locally.

## 

Reply to this email directly or view it on
GitHubhttps://github.com/stratosphere/stratosphere/pull/636#issuecomment-39379590
.

**Comment by rmetzger** [comment]
Okay, great. The pull request is ready to merge in my opinion.

I would really like to merge this rather soon since @alexff91 put a lot of work into it and the sooner we have the checkstyle in place, the better for our code quality.

**Comment by rmetzger** [comment]
I rebased @alexff91's pull request to the current master. I had 6 conflicts which were easy to fix (mainly imports and some if statements).
But there were some checkstyle errors in code that we recently merged.
https://github.com/rmetzger/stratosphere/tree/checkstyle_rebased

I will give you a few more hours to comment on this pull request. (then, I have waited for more than 36 hours and I would like to merge it).

**Comment by alexff91** [comment]
Hi @rmetzger, what I need to do?

**Comment by rmetzger** [comment]
Nothing ;)
Wait until I merge your pull request.

On Fri, Apr 4, 2014 at 1:50 PM, Aleksandr notifications@github.com wrote:

> Hi @rmetzger https://github.com/rmetzger, what I need to do?
> 
> ## 
> 
> Reply to this email directly or view it on GitHubhttps://github.com/stratosphere/stratosphere/pull/636#issuecomment-39556752
> .

## 

Robert Metzger, Kontakt: metzgerr@web.de, Mobil: 0171/7424461

**Comment by alexff91** [comment]
Okey :blush:

**Comment by rmetzger** [comment]
Is anybody against merging this pull request?
If not, I would rebase it again and then finally merge it.

**Comment by fhueske** [comment]
:+1:

**Comment by rmetzger** [comment]
Hey @alexff91, a quick question: how much time do you need to re-apply the formattings to the current `master` ?

I'm really sorry that we handle this pull request so badly.

**Comment by alexff91** [comment]
I think not so much time, I'll do it tonight.

**Comment by rmetzger** [comment]
Is it like 10 minutes or 2 hours?

Can you wait with the re-application of the changes until I give you the go? I think I do not find time to review your changes until Monday, so its probably better to wait until then.

**Comment by alexff91** [comment]
It's like 15-20 minutes :) Ok, I'll wait.

**Comment by rmetzger** [comment]
Do you have skype, google hangout or something like this?
(We recently started using gitter, its something like a web-based IRC with GitHub integration: https://gitter.im/stratosphere/stratosphere) 

I think this would make coordination much easier.

**Comment by alexff91** [comment]
Does gitter has a notifications only when page is opened? Google hangouts I think will be more appropriate(because of phone and chrome-plugin), so you can write me in hangout to alexff91@gmail.com adress.

**Comment by rmetzger** [comment]
Just as an update for the others: I'm working on this issue together with @alexff91, but we are blocked by some Maven issues.

**Comment by fhueske** [comment]
Great, looking forward to have this merged!

**Comment by rmetzger** [comment]
Fixed by https://github.com/stratosphere/stratosphere/pull/751

**Pull request closed without merging**